templates/default/hdfs-site.xml.erb

```diff
@@ -16,7 +16,7 @@
 
   <property>
     <name>dfs.namenode.rpc-address</name>
-    <value><%= @nn_rpc_address %></value>
+    <value><%= node['fqdn'] %>:<%= node['hops']['nn']['port'] %></value>
```

Comment:
See below: This is not going to work if you take a snapshot of a VM for the AMI.

Reply:
See comment above